crates/core/src/geoarrow/mod.rs

```diff
@@ -3,58 +3,33 @@
 pub mod json;
 
 use crate::{Error, ItemCollection, Result};
-use arrow::array::{GenericByteArray, RecordBatch, cast::AsArray, types::GenericBinaryType};
-use arrow::datatypes::{DataType, Field, SchemaBuilder, TimeUnit};
-use arrow::json::ReaderBuilder;
+use arrow_array::{RecordBatch, RecordBatchIterator, RecordBatchReader, cast::AsArray};
+use arrow_json::ReaderBuilder;
+use arrow_schema::{SchemaBuilder, SchemaRef};
 use geo_types::Geometry;
-use geoarrow::{
-    ArrayBase,
-    array::{CoordType, GeometryBuilder, NativeArrayDyn, WKBArray},
-    datatypes::NativeType,
-    table::Table,
+use geoarrow_array::{
+    GeoArrowArray, GeoArrowType,
+    array::{WkbArray, from_arrow_array},
+    builder::GeometryBuilder,
 };
+use geoarrow_schema::{CoordType, GeometryType, Metadata};
 use serde_json::{Value, json};
 use std::sync::Arc;
 
-const DATETIME_COLUMNS: [&str; 8] = [
-    "datetime",
-    "start_datetime",
-    "end_datetime",
-    "created",
-    "updated",
-    "expires",
-    "published",
-    "unpublished",
-];
```

Comment:
Acktwelly I do need to keep these datetime column mappings, so not as close as I thought.